packages/microsite/src/cli/microsite-build.ts

```diff
@@ -557,10 +569,79 @@ async function bundlePagesForSSR(paths: string[]) {
         entry.hydrateBindings = null;
       if (entry.hydrateStyleBindings.length === 0)
         entry.hydrateStyleBindings = null;
+      if (entry.hydrateBindings) {
+        for (const [file, exports] of Object.entries(entry.hydrateBindings)) {
+          const hydrated = hydrateMap[file.replace(/\.js$/, "")];
+          entry.hydrateBindings[file] = Object.keys(exports).reduce(
+            (acc, key) => {
+              return Object.assign(acc, { [hydrated[key] || key]: key });
+            },
+            {}
+          );
+        }
+      }
       return entry;
     });
 }
 
+const rewriteHydratedComponentDisplayNames = (): Plugin => {
+  let withHydrateMap: Record<string, Record<string, string>> = {};
+  return {
+    name: "@microsite/rollup-rewrite-hydrated-component-display-names",
+    renderChunk(code: string, chunk: RenderedChunk) {
+      if (chunk.name.startsWith("_hydrate/")) {
+        const ast = this.parse(code);
+        visit(ast, {
+          visitCallExpression(path) {
+            if (path.get("callee").getValueProperty("name") === "withHydrate") {
+              const exportName = path.parent.get("id").getValueProperty("name");
```

Comment:
I suspect this may not always be the export name.

I'm thinking of this case:
```
const A = withHydrate(Inner)
export {A as B}
```

However, we can iterate on this later as this PR still fixes 99% of cases.

Reply:
I was just thinking about that!

Reply:
Woah, turns out Rollup is super smart and this is never a problem! The local identifier will always match the exported name by this point because Rollup automatically flattens any `x as y` statements.